test/OpenTelemetry.Exporter.Geneva.Tests/UnixUserEventsDataTransportTests.cs

```diff
@@ -0,0 +1,321 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+#if NET6_0_OR_GREATER
+
+#nullable enable
+
+using System.Diagnostics;
+using System.Globalization;
+using System.Text.RegularExpressions;
+using Microsoft.LinuxTracepoints.Provider;
+using OpenTelemetry.Tests;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace OpenTelemetry.Exporter.Geneva.Tests;
+
+[Trait("CategoryName", "Geneva:user_events:metrics")]
+public class UnixUserEventsDataTransportTests
+{
+    /*
+     * Instructions for running these tests:
+     *
+     *  1) You need a version of Linux with user_events available in the kernel.
+     *     This can be done on WSL2 using the 6.6+ kernel.
+     *
+     *  2) You have to run the tests with elevation. You don't need elevation to
+     *     write/emit user_events but you need elevation to read them (which
+     *     these tests do).
+     *
+     *  Command:
+     *    sudo dotnet test --configuration Debug --framework net8.0 --filter CategoryName=Geneva:user_events:metrics
+     *
+     * How these tests work:
+     *
+     *  1) The tests make sure the otlp_metrics tracepoint is registered.
+     *
+     *  2) A process is spawned to run cat /sys/kernel/debug/tracing/trace_pipe. This is what is listening for events.
+     *
+     *  3) Depending on the test, a process is spawned to run sh -c "echo '1' > /sys/kernel/tracing/events/user_events/{this.name}/enable" to enable events.
+     *
+     *  4) The thread running the tests writes to user_events using the GenevaExporter code. Then it waits for a bit. Then it checks to see what events (if any) were emitted.
+     *
+     *  5) Depending on the test, a process is spawned to run sh -c "echo '0' > /sys/kernel/tracing/events/user_events/{this.name}/enable" to disable events.
+     */
+
+    private static readonly byte[] testRequest = [0x0a, 0x0f, 0x12, 0x0d, 0x0a, 0x0b, 0x0a, 0x09, 0x54, 0x65, 0x73, 0x74, 0x4d, 0x65, 0x74, 0x65, 0x72];
+    private readonly ITestOutputHelper testOutputHelper;
+
+    public UnixUserEventsDataTransportTests(ITestOutputHelper testOutputHelper)
+    {
+        this.testOutputHelper = testOutputHelper;
+    }
+
+    [SkipUnlessPlatformMatchesFact(TestPlatform.Linux, requireElevatedProcess: true)]
+    public void UserEvents_Enabled_Succes_Linux()
```

Comment:
@lalitb Something we can steal for OTel Rust's user-events exporters too.

Reply:
Testing with actual user_events tracepoint - yes something we can add.